### PR TITLE
add responsive props section to Flex documentation

### PR DESCRIPTION
**Pull request by clxandstuff**
- added responsive props section in Flex
- added general responsive props section in foundations(we can discuss if this is good place)

### CODE REVIEW

File: src/components/flex/Flex.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+- [Stories](#stories)
+- [Accesibility](#accessibility)
+- [Responsive props](#responsive-props)
+
+## Overview
+
+<Canvas>
+  <Story name="Default">
+    {args => (
+      <Flex>
```
[the remark this answers is not in the document]
Comment (clxandstuff) [reply src/components/flex/Flex.stories.mdx:232]: ah I forgot it, thanks

[the remark this answers is not in the document]
Comment (clxandstuff) [reply src/components/flex/Flex.stories.mdx:232]: fixed

**Comment by katarzynatobis** [comment]
Section for margins is a little bit misleading. This is how it looks like for xxxl:
Screenshot 2022-04-01 at 16 40 35
I would expect that after changing the margin control I would get the same thing, and the effect is completely different:
Screenshot 2022-04-01 at 16 40 19

File: src/components/flex/Flex.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+      },
+    },
+  }}
+/>
+
+# Flex
+
+- [Stories](#stories)
+- [Accesibility](#accessibility)
+- [Responsive props](#responsive-props)
```
Comment (katarzynatobis) [inline src/components/flex/Flex.stories.mdx:225]: Responsive props section is before Accessibility

**Review by katarzynatobis — CHANGES_REQUESTED** [review]

File: src/components/flex/Flex.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+
+## Overview
+
+<Canvas>
+  <Story name="Default">
+    {args => (
+      <Flex {...args}>
+        <Box {...boxStyle} style={{marginRight: 10}}>
```
[the remark this answers is not in the document]
Comment (clxandstuff) [reply src/components/flex/Flex.stories.mdx:233]: > Section for margins is a little bit misleading. This is how it looks like for xxxl: Screenshot 2022-04-01 at 16 40 35 I would expect that after changing the margin control I would get the same thing, and the effect is completely different: Screenshot 2022-04-01 at 16 40 19

Margin is outside Flex so for default story which renders single Flex is actually ok. I think, I know how I can improve story itself.

[the remark this answers is not in the document]
Comment (clxandstuff) [reply src/components/flex/Flex.stories.mdx:233]: @katarzynatobis take a look please, I changed box labels, and Margins story description.

File: src/components/flex/Flex.stories.mdx (hunk 2 above)
Comment (clxandstuff) [reply src/components/flex/Flex.stories.mdx:224]: @katarzynatobis Done

**Comment by clxandstuff** [comment]
@katarzynatobis @coderitual
I added gray box for flex contrainer. Please check this out 😄

**Review by katarzynatobis — APPROVED** [review]

**Pull request merged**